core/runtime/src/main/java/io/quarkus/runtime/CleanableExecutor.java

```diff
@@ -34,12 +34,7 @@
     private final EnhancedQueueExecutor executor;
 
     private static final AtomicInteger generation = new AtomicInteger(1);
-    private final ThreadLocal<Integer> lastGeneration = new ThreadLocal<Integer>() {
-        @Override
-        protected Integer initialValue() {
-            return -1;
-        }
-    };
+    private static final ThreadLocal<Integer> lastGeneration = ThreadLocal.withInitial(() -> -1);
```

Comment:
I disagree with that one. We want to avoid lambdas in runtime code.

Reply:
fixed